internal/pkg/service/buffer/store/watcher.go

```diff
@@ -0,0 +1,68 @@
+package store
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+)
+
+type Watcher struct {
+	mappings         sync.Map
+	secrets          sync.Map
+	slicesForExports sync.Map
+	store            *Store
+}
+
+func NewWatcher(store *Store) (*Watcher, error) {
+	w := &Watcher{
+		mappings:         sync.Map{},
+		secrets:          sync.Map{},
+		slicesForExports: sync.Map{},
```

Comment:
Vím že jsme to na začátku navrhovali jinak (např. i nějaký ten `ReceiverKeyWithSecret`), ale nakonec jsem to změnil tak, aby se odtud daly potřebný data do importu co nejsnáz natáhnout.

Reply:
Tady to vlastně kvůli tý sync mapě není vidět, ty typy jsou:
```
mappings: map[key.ReceiverKey]map[key.ExportKey]*model.Mapping
secrets: map[key.Receiver]string
slicesForExports: map[key.ExportKey]time.Time
```

Reply:
Uplne v pohode, nemal som to domyslene do konca, odovzdal som iba to, co som mal zatial nachystane.
Keby si to `ReceiverKeyWithSecret` v kode nechal, tak by som ti to navrhol dat prec :) ako som nad tym dalej rozmyslal. Jednoduchsie je ukladat to ako `ReceiverKey` a ten secret skontrolovat podmienkou.